addons/binding/org.openhab.binding.cm11a/src/main/java/org/openhab/binding/cm11a/handler/Cm11aLampHandler.java

```diff
@@ -0,0 +1,179 @@
+                if (OnOffType.ON.equals(command)) {
+                    desiredState = OnOffType.ON; // PercentType.HUNDRED;
+                } else if (OnOffType.OFF.equals(command)) {
+                    desiredState = OnOffType.OFF; // PercentType.ZERO;
+                } else if (command instanceof PercentType) {
+                    desiredState = (PercentType) command;
+                } else if (command instanceof RefreshType) {
+                    logger.info("Received REFRESH command for switch " + houseUnitCode);
+                } else {
+                    logger.error("Ignoring unknown command received for device: " + houseUnitCode);
+                }
+
+                if (!(desiredState instanceof UnDefType)) {
+                    X10Interface x10Interface = cm11aHandler.getX10Interface();
+                    x10Interface.scheduleHWUpdate(this);
+                }
+            } else {
+                logger.error("Attenpted to change switch " + houseUnitCode + " cm11a is not online");
+            }
+        } else {
+            logger.error(
+                    "Attenpted to change switch " + houseUnitCode + " but the cm11 module has not bee loaded yet.");
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.openhab.binding.cm11a.handler.Cm11aAbstractHandler#updateHardware(org.openhab.binding.cm11a.internal.
+     * X10Interface)
+     */
+    @Override
+    public void updateHardware(X10Interface x10Interface) throws IOException, InvalidAddressException {
+
+        if (!desiredState.equals(currentState)) {
+            try {
+                boolean x10Status = false;
+                if (desiredState.equals(OnOffType.ON)) {
+                    x10Status = x10Interface.sendFunction(houseUnitCode, X10Interface.FUNC_ON);
+                } else if (desiredState.equals(OnOffType.OFF)) {
+                    x10Status = x10Interface.sendFunction(houseUnitCode, X10Interface.FUNC_OFF);
+                } else {
+                    // desiredState must be a PercentType if we got here.
+                    // Verify the type and calc how many bright increments (0 to 22) we need to send
+                    if (desiredState instanceof PercentType) {
+                        // Calc how many bright increments we need to send (0 to 22)
+                        int desiredPercentFullBright = ((PercentType) desiredState).intValue();
+                        int dims = (desiredPercentFullBright * DIM_LEVELS) / 100;
+                        if (currentState.equals(OnOffType.ON)) {
+                            // The current level isn't known because it would have gone to
+                            // the same level as when last turned on. Need to go to full dim and then up to desired
+                            // level.
+                            x10Interface.sendFunction(houseUnitCode, X10Interface.FUNC_DIM, DIM_LEVELS);
```

Comment:
`DIM_LEVELS` sounds like something else, for example a set. Maybe call it `MAX_DIM_LEVEL`